tests/testsuite/lockfile_path.rs

```diff
@@ -400,6 +400,56 @@ bar = "0.1.0"
     assert_e2e().eq(contents, lockfile_original);
 }
 
+#[cargo_test]
+fn install_not_supported() {
+    Package::new("bar", "0.1.0").publish();
+    Package::new("bar", "0.1.1")
+        .file("src/lib.rs", "not rust")
+        .publish();
+    Package::new("foo", "0.1.0")
+        .dep("bar", "0.1")
+        .file("src/lib.rs", "")
+        .file(
+            "src/main.rs",
+            "extern crate foo; extern crate bar; fn main() {}",
+        )
+        .file(
+            "Cargo.lock",
+            r#"
+[[package]]
+name = "bar"
+version = "0.1.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+
+[[package]]
+name = "foo"
+version = "0.1.0"
+dependencies = [
+ "bar 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
+]
+"#,
+        )
+        .publish();
+
+    cargo_process("install foo")
+        .with_stderr_data(str![[r#"
+...
+[..]not rust[..]
+...
+"#]])
+        .with_status(101)
+        .run();
+    cargo_process("install foo -Zunstable-options --lockfile-path lockfile_dir/Cargo.lock")
+        .masquerade_as_nightly_cargo(&["lockfile-path"])
+        .with_stderr_data(str![[r#"
+...
+[ERROR] unexpected argument '--lockfile-path' found
+...
+"#]])
+        .with_status(1)
+        .run();
```

Comment:
The intent behind splitting the commits is to show the change in behavior.

Showing the flag is not supported is the less interesting change in behavior.  I believe the intent would be to have the test without the flag in this commit, showing what behavior you get today.  The next commit would add the flag and show how the behavior changed.

Reply:
Hmm okay I see what you are saying so we want to pass just --locked without --lockfile-path on the first commit and then change the behavior. Let me do that!

Reply:
✅